models/camelot/camelot_pair_fee_rates.sql

```diff
@@ -0,0 +1,169 @@
+    ),
+    v3_pairs_with_initial_fee_rates as (
+        select
+            date_trunc('minute', evt_block_time) as minute,
+            pool as pair,
+            '3' as version,
+            token0,
+            {{ v3_default_fee }} / {{ v3_fee_precision }} as token0_fee_rate,
+            token1,
+            {{ v3_default_fee }} / {{ v3_fee_precision }} as token1_fee_rate
+        from {{ source("camelot_v3_arbitrum", "AlgebraFactory_evt_Pool") }}
+        {% if not is_incremental() %}
+            where evt_block_time >= timestamp '{{project_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+            where evt_block_time >= date_trunc('day', now() - interval '7' day)
+        {% endif %}
+    ),
+    v3_directional_fee_rate_updates as (
+        select
+            date_trunc('minute', evt_block_time) as minute,
+            pair,
+            version,
+            token0,
+            avg(feezto) / {{ v3_fee_precision }} as token0_fee_rate,  -- Handle edge case where pair fees gets changed multiple times per minute
+            token1,
+            avg(feeotz) / {{ v3_fee_precision }} as token1_fee_rate  -- Handle edge case where pair fees gets changed multiple times per minute
+        from {{ source("camelot_v3_arbitrum", "AlgebraPool_evt_Fee") }} as fee_updates
+        join
+            v3_pairs_with_initial_fee_rates as pairs
+            on fee_updates.contract_address = pairs.pair
+        group by date_trunc('minute', evt_block_time), pair, version, token0, token1
+    ),
+    pairs as (
+        select *
+        from v2_pairs_with_initial_fee_rates
+        union all
+        select *
+        from v3_pairs_with_initial_fee_rates
+    ),
+    fee_rate_updates as (
+        select *
+        from pairs
+        union all
+        select *
+        from v2_directional_fee_rate_updates
+        union all
+        select *
+        from v3_directional_fee_rate_updates
+    ),
+    -- This approach does not work: Result of sequence function must not have more
+    -- than 10000 entries
+    /*time_series as (
+        select date_trunc('minute', block_time) as minute
+        from
+            (
+                select
+                    sequence(
+                        timestamp '{{project_start_date}}'
+                        cast(current_timestamp as timestamp),
+                        interval '1' minute
+                    ) as timestamp_array
+            )
+        cross join unnest(timestamp_array) as t(block_time)
+    ),*/
+    time_series as (
+        select distinct date_trunc('minute', block_time) as minute
+        from dex.trades
+        where block_time >= timestamp '{{project_start_date}}'
+    ),
+    -- Prepare data structure (1 row for every minute since each pair was created)
+    pairs_by_minute as (
+        select time_series.minute, pair, version, token0, token1
+        from pairs
+        cross join time_series
+        where time_series.minute >= pairs.minute
+    )
```

Comment:
as discussed in discord, you are hitting a timeout error on CI (90 minutes). my best guess is this portion of the code. please take the time to figure out where the query really slows down and becomes inefficient. i wouldn't expect data on arbitrum to take this long.

Reply:
yeah probably this cross join, or the last_value() window function a bit further down the code.
but this was the only possibility to build the query imo, do you have suggestions how it could be optimized?

if something is unclear about the code's intentions, pls lmk!

Reply:
@jeff-dude as you can see in the #5862 PR, the bottleneck is in the `last_value()` window function in the final query statement, not in the cross join

So we should probably focus on this, any thoughts on how to improve it would be very welcome 💯